src/librbd/api/Mirror.cc

```diff
@@ -2745,14 +2767,22 @@ int Mirror<I>::group_enable(IoCtx& group_ioctx, const char *group_name,
   cls::rbd::MirrorMode mirror_mode;
   r = cls_client::mirror_mode_get(&group_ioctx, &mirror_mode);
   if (r < 0) {
-    lderr(cct) << "cannot enable mirroring, failed to retrieve mirror mode: "
+    lderr(cct) << "cannot enable mirroring, failed to retrieve mirror mode for: "
+               << Mirror<I>::gpool_or_gnamespace(group_ioctx) << ", :"
                << cpp_strerror(r) << dendl;
     return r;
   }
 
+  if (mirror_mode == cls::rbd::MIRROR_MODE_DISABLED ||
```

Comment:
This implies cls::rbd::MIRROR_MODE_POOL is valid. Should it check for cls::rbd::MIRROR_MODE_IMAGE instead?